tests/library/inspector/cli-codegen-3.spec.ts

```diff
@@ -559,6 +559,60 @@ await page.GetByLabel("Coun\\"try").ClickAsync();`);
     ]);
   });
 
+  test('should consume contextmenu events, despite a custom context menu', async ({ page, openRecorder }) => {
+    const recorder = await openRecorder();
+
+    await recorder.setContentAndWait(`
+      <button>Right click me.</button>
+      <div id="menu" style="display: none; position: absolute;">
+        <button>Menu option 1</button>
+        <button>Menu option 2</button>
+      </div>
+      <script>
+        const button = document.querySelector('button');
+        button.addEventListener('contextmenu', e => {
+          e.preventDefault();
+          console.log('right-clicked');
+
+          // show custom context menu
+          const menu = document.getElementById("menu");
+          menu.style.display = "block";
+          menu.style.left = \`\${e.pageX}px\`;
+          menu.style.top = \`\${e.pageY}px\`;
+        });
+        const log = [];
+        for (const eventName of ['mousedown', 'mousemove', 'mouseup', 'pointerdown', 'pointermove', 'pointerup', 'click', 'contextmenu']) {
+          button.addEventListener(eventName, e => log.push('button: ' + e.type));
+          menu.addEventListener(eventName, e => log.push('menu: ' + e.type));
+        }
+      </script>
+    `);
+
+    await recorder.hoverOverElement('button');
+    expect(await page.evaluate('log')).toEqual(['button: pointermove', 'button: mousemove']);
+
+    const [message] = await Promise.all([
+      page.waitForEvent('console', msg => msg.type() !== 'error'),
+      recorder.waitForOutput('JavaScript', `button: 'right'`),
+      recorder.trustedClick({ button: 'right' }),
+    ]);
+    expect(message.text()).toBe('right-clicked');
```

Comment:
When clicking manually on the page, the contextmenu listener is not invoked for me when recorder is on. It does get invoked without the recorder. Do you know what's going on? I'm on macbook pro, and doing ctrl+click on the trackpad.

Reply:
Thank you for finding this issue. Here are some observations (with PR active):

![CleanShot 2024-07-15 at 16 20 55](https://github.com/user-attachments/assets/8d98090a-4fbd-45c8-a153-5624b4c6f31d)

I will keep investigating.

Reply:
The following commit covers all three cases: https://github.com/microsoft/playwright/pull/31634/commits/5028769b527129a0722e5b9949f4bdf1ea2d2c09

1. Ctrl + click
2. Two finger click
3. External mouse right click

by having Playwright perform a right click after consuming `contextmenu`.

The previous commit did not cover the `ctrl + click` case because Playwright's performance of `ctrl + click` did not cause a `contextmenu` event, for some reason.